pkg/dockerutil/volumes.go

```diff
@@ -0,0 +1,141 @@
+package dockerutil
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	ddevImages "github.com/ddev/ddev/pkg/docker"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/util"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/volume"
+)
+
+// RemoveVolume removes named volume. Does not throw error if the volume did not exist.
+func RemoveVolume(volumeName string) error {
+	ctx, client, err := GetDockerClient()
+	if err != nil {
+		return err
+	}
+	if _, err := client.VolumeInspect(ctx, volumeName); err == nil {
+		err := client.VolumeRemove(ctx, volumeName, true)
+		if err != nil {
+			if err.Error() == "volume in use and cannot be removed" {
```

Comment:
No need to touch this here (way off topic) but wow we need to do fancier things with this instead of looking for a string in the returned error. Also, this had led to some very complicated support situations because we don't seem to adequately report it. However, the code below seems to attempt clearly to handle this.